personio-to-sheets/PersonioDump.js

```diff
@@ -485,3 +485,35 @@ function writeRelationsToSheet_(spreadsheetId, relations) {
 
     Sheets.Spreadsheets.batchUpdate(batch, spreadsheetId);
 }
+
+
+/** Get a spreadsheet by ID. */
+function getSpreadsheet_(id) {
+    const spreadsheet = Sheets.Spreadsheets.get(id);
+    if (!spreadsheet)
+        throw new Error(`Specified spreadsheet ${id} does not exist or is not accessible`);
+
+    return spreadsheet;
+}
+
+
+/** Creates a "custom" sheet with the specified name (if it doesn't exist) and returns the sheet properties. */
+function ensureSheet_(spreadsheet, sheetTitle) {
```

Comment:
Why can the generic ensureSheet not be used here? Maybe something to refactor in a follow up?

Reply:
The generic `SheetUtil.ensureSheet()` uses the basic `SpreadsheetApp` API.

Some problems require direct use of the Google Sheets API via HTTP requests or the "Advanced Sheets Service" (a special wrapper for the Google Sheets API) to make use of batching, access to certain properties and to keep performance high and the number of individual service requests low (to avoid running into quotas).

The Personio to sheets export is a use-case benefitting from the use of the "Advanced Sheets Service".

We still want the general utility functions to be available to other scripts to reduce the amount of duplicated code.

Reply:
The problem here is, that there are 3 different, non-interoperable (different objects) ways to work with the Google Sheets API.

The `SheetUtil` utility functions should use the most common, basic API.